cmd/object-api-interface.go

```diff
@@ -114,29 +130,25 @@ func (r *HashReader) Read(p []byte) (n int, err error) {
 // data.
 func (r *HashReader) Size() int64 { return r.size }
 
-// MD5 returns the MD5 sum of the processed data. Any
-// further reads will change the MD5 sum.
-func (r *HashReader) MD5() []byte { return r.md5Hash.Sum(nil) }
-
-// Verify verifies if the computed MD5 sum - and SHA256 sum - are
+// verify verifies if the computed MD5 sum - and SHA256 sum - are
 // equal to the ones specified when creating the HashReader.
-func (r *HashReader) Verify() error {
+func (r *HashReader) verify() error {
 	if r.sha256Hash != nil {
-		sha256Sum, err := hex.DecodeString(r.sha256Sum)
+		sha256Sum, err := hex.DecodeString(r.SHA256Sum)
```

Comment:
This check should be in `NewHashReader()` function.

Reply:
Yeah NewHashReader is not implemented in this manner.. @balamurugana that would require a larger change.

Reply:
I guess its related fix only because `HashReader` does verification now which changes the usability.